### ISSUE TITLE
AppVeyor remote feed package open fails with "Object reference not set to an instance of an object" message

### ISSUE THREAD

**Issue by tker** [NuGetPackageExplorer/NuGetPackageExplorer#386]
Hi! Thanks for reporting this feature/bug/question! 

Please keep / fill in the relevant info from this template so that we can help you as best as possible.

**Type**:

- Bug

**NPE version**: 4.2.13+g3c913990f30

**Installed from**: chocolatey.org  and Windows Store (tried both, currently using Windows Store installed version)

In case of a BUG:

- What is the current result?
Every time I try to open a package from our private AppVeyor feed the package fails to either open or download with an "Object reference not set to an instance of an object" exception. 

Packages and versions list correctly; they simply fail to open.

- What is the expected result?

Package should open or download as it does for NuGet.org packages.

- Please post full exception details in case of an Exception (message, stacktrace, inner exceptions).

"Object reference not set to an instance of an object"

- Are there any workarounds? yes/no

Can retrieve package via other methods then open as local package.

- Is there a version in which it did work?

Don't know.

- Is there a package on Nuget.org which shows the bug?

Not that I am aware of.

![image](https://user-images.githubusercontent.com/1005753/38490127-cc3492c2-3c1a-11e8-9935-65abf5e657d0.png)

**Comment by tker** [comment]
Extra information: I am running NuGet.exe Version: 4.6.2.5055, but this issue also occurred with previously installed 4.5.x .

My Windows is set set to automatically update, current system information is:

![image](https://user-images.githubusercontent.com/1005753/38490517-e2105e86-3c1b-11e8-9bfb-51c12541526b.png)

Local packages open perfectly as do remote packages from NuGet.org and AppVeyor feed works perfectly in all other situations.

**Comment by campersau** [comment]
Thanks for the report.
I was able to reproduce it, I will look into it!

**Comment by tker** [comment]
Thank you! Thanks for the quick response too.

**Comment by campersau** [comment]
#387 should fix the issue.
You can try it out with the builds from https://ci.appveyor.com/project/NuGetPackageExplorer/nugetpackageexplorer/build/4.3.11+gc6ff541696/artifacts

I also realized that the credential caching does not work properly. I will look into it too.

**Comment by tker** [comment]
That is brilliant, thank you very much! 

Works like a dream. I shall keep an eye out for the credential caching fix: would you like me to close the issue or is that something you will do?

**Comment by campersau** [comment]
👍 

No you don't have to do anything. This issue will be closed automatically once #387 is merged.

I have opened a new issue for the credential caching bug https://github.com/NuGetPackageExplorer/NuGetPackageExplorer/issues/388
